impl/client/src/main/javascript/web/pentaho/data/filter/_core/isLike.js

```diff
@@ -0,0 +1,128 @@
+/*!
+ * Copyright 2017 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../KnownFilterKind"
+], function(KnownFilterKind) {
+
+  "use strict";
+
+  return function(filter) {
+
+    /**
+     * @name pentaho.data.filter.IsLike.Type
+     * @class
+     * @extends pentaho.data.filter.Property.Type
+     *
+     * @classDesc The type class of the is-like filter type.
+     *
+     * For more information see {@link pentaho.data.filter.IsLike}.
+     */
+
+    /**
+     * @name pentaho.data.filter.IsLike
+     * @class
+     * @extends pentaho.data.filter.Property
+     * @amd {pentaho.type.Factory<pentaho.data.filter.IsLike>} pentaho/data/filter/isLike
+     *
+     * @classDesc The `IsLike` class represents a simple string matching filter.
+     * This filter selects elements having the formatted value match an affix
+     * (ie prefix, suffix or infix) of a reference value: [value]{@link pentaho.data.filter.IsLike#value}.
```

Comment:
the infix case is the only one we directly support in DET :) prefix and suffix are supported only as the degenerate case of the infix. However, this is a general filter in common ui, there is no reason not to explicitly support all cases.